### PR TITLE
Document and perform manually generating release notes

### PR DESCRIPTION
**Pull request by tisonkun**
This closes https://github.com/apache/pulsar/issues/13207 as discard.

1. Release note generation is a rare job. The current `release-note-gen.sh` scripts only update the metadata - the release manager should still check in the release note manually: https://github.com/apache/pulsar-site/pull/262
2. It's unclear how the build process works and cannot preview locally.
3. Now, Pulsar Client C++ is moved to a new repo https://github.com/apache/pulsar-client-cpp and requires a new process. But the script is so brittle and coupled to extend for new requirement.
4. Moreover, the current solution cause generated failure on Client Python and Client Websocket for 2.10.2. I strongly suggest we do it manually, simple, stupid, and correct: https://pulsar.apache.org/release-notes/#python
    * WebSocket 2.10.1 doesn't have a 2.10.x version tag.
    * Python 2.10.2 directs to 404.

So,

1. Remove the `release-note-gen.sh` and related stuff.
2. Manually generate release notes. You can find it easy to perform.
3. Update the contribution guide.

cc @BewareMyPower @urfreespace @michaeljmarshall @merlimat @Anonymitaet 

to @BewareMyPower @merlimat please check whether it's clear now how to add the 3.0.0 release note. I think it's simply:

1. Add the release note content to `site2/website-next/release-notes/versioned/client-cpp-3.0.0.md`
2. Update the metadata at `site2/website-next/data/release-cpp.js` follow the surrounding pattern.

### CODE REVIEW

**Review by BewareMyPower — APPROVED** [review]
The preview LGTM.

image

image

But have a question about the preview. In my local preview, the `/docs/client-libraries-cpp` path shows "Page Not Found". I think it might need the external documents from the `apache/pulsar` repo, but is there a way to make it visible in local preview?

**Comment by urfreespace** [comment]
Hi @SignorMercurio would you pls take a look at this PR, thanks.

**Comment by Anonymitaet** [comment]
> 2. It's unclear how the build process works and cannot preview locally.

@SignorMercurio can you explain the current build process of release note auto-generation?

**Comment by tisonkun** [comment]
@BewareMyPower By default, preview builds only the `next` version. To meet your requirement, run `./preview.sh 2.10.x`.

**Comment by tisonkun** [comment]
> > 3\. It's unclear how the build process works and cannot preview locally.
> 
> @SignorMercurio can you explain the current build process of release note auto-generation?

@Anonymitaet No. It's not auto generated. The metadata is built when deploy the site, the release notes content must be checked in manually.

**Review by urfreespace — APPROVED** [review]

**Comment by Anonymitaet** [comment]
Oh, I see, thanks for your explanations! We can try this new approach first. It looks more simple and friendly for release managers.

File: site2/website-next/contribute/releasing/release-note-guide.md [hunk 1]
```diff
@@ -12,22 +12,35 @@ This chapter gives an overview of Pulsar release notes.
...
+
+```bash
+# Replace v2.10.2 with the target version tag
+gh release view "v2.10.2" -R apache/pulsar --json body --jq .body
+```
+
+3. Select the related release notes entries and add a [versioned release note](https://github.com/apache/pulsar-site/tree/main/site2/website-next/release-notes/versioned).
+4. Update the metadata file [site2/website-next/data/release-*.js](https://github.com/apache/pulsar-site/tree/main/site2/website-next/data).
+
+The process is similar to update standalone client releases.
```
Comment (Anonymitaet) [inline site2/website-next/contribute/releasing/release-note-guide.md:43]: Can we add the workflow of "previewing release note" to line 67? Can we reuse https://github.com/apache/pulsar-site/blob/main/site2/website-next/contribute/documentation/preview.md#preview-website-changes? If yes, just add a link there

Comment (tisonkun) [reply site2/website-next/contribute/releasing/release-note-guide.md:43]: Thanks for your hint! Make sense and updated 224a9c846aed9d0735a82219cd9d85709eb6a0a2.

**Review by Anonymitaet — APPROVED** [review]

**Pull request merged**